crates/polars-python/src/series/scatter.rs

```diff
@@ -31,6 +31,25 @@ impl PySeries {
 fn scatter(mut s: Series, idx: &Series, values: &Series) -> Result<Series, (Series, PolarsError)> {
     let logical_dtype = s.dtype().clone();
 
+    let values = if logical_dtype.is_categorical() || logical_dtype.is_enum() {
```

Comment:
Huh?

Reply:
There are probably other issues with `scatter` but the problem here is that it blindly casts something to the physical before scattering. So if you were to scatter floats into a categorical it would first blindly cast the floats into integers, then scatter those integers as physical categories into the categorical series...